/**
 * Copyright (C) 2011 Adriano Monteiro Marques
 *
 * Author:  Zhongjie Wang <wzj401@gmail.com>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
 * USA
 */

package org.umit.icm.proto;

message ICMReport {
    required int64 reportID = 1;
    required int64 senderID = 2;
    required int64 timestamp = 3;
    required int32 testID = 4;
    repeated bytes passedNode = 5;	// the peer in the way of routing to the aggregator, this field is repeated
    // report details
    repeated WebsiteReportDetail websiteReport = 100;
    repeated ServiceReportDetail serviceReport = 101;
    // extensions 100 to max;
}

/*
extend ICMReport {
    repeated WebsiteReportDetail websiteReport = 100;
    repeated ServiceReportDetail serviceReport = 101;
}
*/

message WebsiteReportDetail {
    required string websiteURL = 2;
    required int32 statusCode = 3;
    optional double responseTime = 4;
    optional double bandwidth = 5;
    extensions 10 to 99;	// can add more fields later
}

message ServiceReportDetail {
    required string serviceName = 2;
    required int32 statusCode = 3;
    optional double responseTime = 4;
    optional double bandwidth = 5;
    extensions 10 to 99;	// can add more fields later
}

